packages/client/src/api/wargames_api/index.ts

```diff
@@ -152,46 +151,64 @@ export const listenForWargameChanges = (name: string, dispatch: PlayerUiDispatch
   listenNewMessage({ db, name, dispatch })
 }
 
-export const pingServer = (activityDetails: { wargame: string, role: string }): Promise<any> => {
-  const activityMissing = 'The player has not shown any activity yet'
-  const activityTime = encodeURIComponent(expiredStorage.getItem(`${activityDetails.role}_${ACTIVITY_TIME}`) || activityMissing)
-  const activityType = encodeURIComponent(expiredStorage.getItem(`${activityDetails.role}_${ACTIVITY_TYPE}`) || activityMissing)
+export const pingServer2 = async (log: ActivityLogsInterface): Promise<string> => { 
+  console.warn('ping server currently commented out') 
+  const items = log.items
+  const first = items[0]
 
-  const activityUrl = `${activityDetails.wargame || 'missing'}/${activityDetails.role || 'missing'}/${activityTime}/${activityType}`
+  // TODO: in addition to pushing data to the server, we're also checking the server is still alive
+  // So, even if the log is empty, we should push an empty list, since we want to get a 
+  // 'success' back from the server
 
-  return fetch(`${serverPath}healthcheck/${activityUrl}/healthcheck`, {
-    method: 'GET',
-    headers: {
-      'Content-Type': 'application/json'
-    }
-  })
-    .then((response: Response): Promise<any> => response.json())
-    .then((data: any) => {
-      return data.status
-    })
-    .catch((err) => {
-      console.log(err)
-      return 'NOT_OK'
-    })
-}
+  // TODO: this method should receive an array of activities.  It should push them all as
+  // documents to the PlayerLogs database.  We don't extend the existing set of documents,
+  // we push them as new documents via Couchdb "bulk push" call.
+  const { db } = getWargameDbByName(log.currentDbname)
 
-export const getPlayerActivityLogs = () => {
-  const url = `${serverPath}playerlog`
+  if (!first) return db.putPlayerLogs(first).then(res => res.msg)
 
-  return fetch(url, {
-    method: 'GET',
-    headers: {
-      'Content-Type': 'application/json'
-    }
-  })
-    .then((response: Response): Promise<any> => response.json())
-    .then((data: any) => {
-      return data
-    })
-    .catch((err) => {
-      console.log(err)
+  const { wargame, role, activityTime, activityType } = first
+
+  return await getPlayerActivityLogs(wargame, log.currentDbname)
```

Comment:
@lilitkarapetyan - this code appends the logs to the existing set for this wargame.

We don't need to do that.  We can just push all the log items to the `player-log` database.  The server API currently just allows one document to be pushed - so we just need to either switch it to taking an array (and insert them as individual docs), or to use a "bulk update" PouchDb API call to push/store all the docs at once.

Reply:
Ok, I will work on it

Reply:
(seem to have lost this comment)

@lilitkarapetyan - we're making these changes because a scientist wants to download the full history of activities for a wargame - not just the most recent one.  So, on the client we are building up an array of activities, and pushing them all to the server on `pingServer`.   That means `player-log` will grow to have all activites for all roles for all wargames.

That's a lot more data.  It also means that when we retrieve the player logs for a wargame, she server needs to only provide the most recent activity (newest `id`) for each role.